### PR TITLE
feat: convert this library to be "buildless"

### PR DESCRIPTION
**Pull request by bradenmacdonald**
There's no reason to compile this code into a `dist` folder before we publish it. It's only used in the Course Authoring MFE, and that MFE's webpack build has no problem consuming this as raw JSX/TSX files. Publishing an untransformed build has a number of advantages: makes development easier, speeds up CI, simplifies things, and applies consistent babel/polyfill settings to all of the code in the MFE, instead of having separate settings for the editors vs. the rest of the MFE.

Part of https://github.com/openedx/frontend-lib-content-components/issues/499

# How to test:

### Simple way:
1. Check out [this commit of `frontend-app-course-authoring`](https://github.com/open-craft/frontend-app-course-authoring/commit/1212797d664b878ac107df02a53391a4f8ed4db9) or edit its `package.json` and change the `@edx/frontend-lib-content-components` version to: `github:open-craft/frontend-lib-content-components#braden/buildless`
2. Run `npm install`
3. Start course-authoring however you normally do (e.g. `APP_ID=course-authoring PUBLIC_PATH=/course-authoring/ MFE_CONFIG_API_URL='http://localhost:8000/api/mfe_config/v1' npm run start -- --host apps.local.overhang.io`)
4. Verify that you can still edit components in courses

### Advanced way

⁉️ Is there an easier way to do this? ⁉️🧐

With this method you can more easily modify the code and get hot reloading as you do:

1. Check out this branch, and make sure this repo's folder is alongside the course authoring MFE's folder.
2. Run the following from the course authoring MFE's folder:
    ```
    npm install
    rm -rf node_modules/@edx/frontend-lib-content-components && ln -s "$(realpath ../frontend-lib-content-components)" node_modules/@edx/frontend-lib-content-components
    cd ../frontend-lib-content-components
    rm -rf node_modules && ln -s ../frontend-app-course-authoring/node_modules node_modules
    ```
3. Start course-authoring however you normally do
4. Verify that you can still edit components in courses, and when you make changes to this repo's code, they're immediately reflected in the course authoring MFE.

# Concerns

The `www` project doesn't work for me, either before or after this PR. Is it supposed to be working??

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @bradenmacdonald!

### What's next?

*Please work through the following steps to get your changes ready for engineering review:*

#### :radio_button: Get product approval

If you haven't already, [check this list](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Does-my-contribution-require-Product-Review%3F) to see if your contribution needs to go through the product review process.

- If it does, you'll need to submit a product proposal for your contribution, and have it reviewed by the [Product Working Group](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3449028609/Product+Working+Group).
    - This process (including the steps you'll need to take) is documented [here](https://openedx.atlassian.net/wiki/spaces/COMM/pages/3875962884/How+to+submit+an+open+source+contribution+for+Product+Review#Product-Review-Process).
- If it doesn't, simply proceed with the next step.

#### :radio_button: Provide context

To help your reviewers and other members of the community understand the purpose and larger context of your changes, feel free to add as much of the following information to the PR description as you can:

- Dependencies
  > This PR must be merged before / after / at the same time as ...
- Blockers
  > This PR is waiting for OEP-1234 to be accepted.
- Timeline information
  > This PR must be merged by XX date **because** ...
- Partner information
  > This is for a course on edx.org.
- Supporting documentation
- Relevant [Open edX discussion forum](https://discuss.openedx.org/) threads





#### :radio_button: Get a green build

If one or more checks are failing, continue working on your changes until this is no longer the case and your build turns green.



#### :radio_button: Let us know that your PR is ready for review:

### Who will review my changes?


This repository is currently maintained by `@openedx/2u-tnl`.  Tag them in a comment and let them know that your changes are ready for review.


### Where can I find more information?

If you'd like to get more details on all aspects of the review process for open source pull requests (OSPRs), check out the following resources:

- [Overview of Review Process for Community Contributions](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/FAQ-about-pull-requests.html)
- [Pull Request Status Guide](https://docs.openedx.org/en/latest/developers/references/developer_guide/process/pull-request-statuses.html)
- [Making changes to your pull request](https://docs.openedx.org/en/latest/documentors/how-tos/make_changes_to_your_pull_request.html)

### When can I expect my changes to be merged?

Our goal is to get community contributions seen and reviewed as efficiently as possible.

However, the amount of time that it takes to review and merge a PR can vary significantly based on factors such as:

- The size and impact of the changes that it introduces
- The need for product review
- Maintenance status of the parent repository

:bulb: *As a result it may take up to several weeks or months to complete a review and merge your PR.*

File: README.md [hunk 1]
```diff
@@ -101,6 +101,19 @@ It will also configure the editor to be viewable in the gallery view. Adding the
 - To make this easier, we provide example files for `jest.config.js` and `setupTest.js` that are known to work.
 You can find them in the example/ folder.
 
+## Installing into Course Authoring MFE in dev mode
```
Comment (bradenmacdonald) [inline README.md:104]: Is there a better way to do this?

**Comment by bradenmacdonald** [comment]
Closing in favor of https://github.com/openedx/frontend-app-course-authoring/pull/1208

**Comment by openedx-webhooks** [comment]
@bradenmacdonald Even though your pull request wasn’t merged, please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-lib-content-components&entry.867055334=https://github.com/openedx/frontend-lib-content-components/pull/498&entry.1484655318=https://github.com/bradenmacdonald&entry.752974735=2024-08-01+22:25&entry.1917517419=2024-08-14+20:54&entry.2133058324=No) so we can improve your experience in the future.

**Pull request closed without merging**